plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -242,86 +300,123 @@ func (p *PrometheusClient) Write(metrics []telegraf.Metric) error {
 			labels[sanitize(k)] = v
 		}
 
-		// Prometheus doesn't have a string value type, so convert string
-		// fields to labels.
-		for fn, fv := range point.Fields() {
```

Comment:
I think this loop can stay, since it shouldn't interact with the different types and its theoretically possible to have a string added to a summary.

Side note: This is a symptom of how the ValueType was added later to our metrics, and how the types really should be per field not per metric.  We will probably try to redo the Metrics class in the next year but for now we have to deal with it.